test/SlowTests/Server/Documents/AI/AiAgent/AiAgentClientApiBasics.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FastTests;
+using FastTests.Client;
+using Orders;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Linq;
+using Raven.Client.Documents.Operations.AI;
+using Raven.Client.Documents.Operations.AI.Agents;
+using Raven.Client.Documents.Operations.ConnectionStrings;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+
+namespace SlowTests.Server.Documents.AI.AiAgent;
+
+public class AiAgentClientApiBasics : RavenTestBase
+{
+    public AiAgentClientApiBasics(ITestOutputHelper output) : base(output)
+    {
+    }
+
+    public class OutputSchema
+    {
+        public string Answer = "Answer to the user question";
+
+        public bool Relevant = true;
+
+        public List<string> RelevantOrdersId = ["The order ids relevant to the query or response"];
+
+        public List<string> MatchingProductsId = ["All the product ids referenced either by the user or the system"];
+    }
+
+    [RavenTheory(RavenTestCategory.Ai)]
+    [RavenGenAiData(IntegrationType = RavenAiIntegration.OpenAi, DatabaseMode = RavenDatabaseMode.Single, CheckCanConnect = false, NightlyBuildRequired = false)]
+    public async Task AiAgentClientApiBasicTest(Options options, GenAiConfiguration config)
+    {
+        using var store = GetDocumentStore(options);
+        await store.Maintenance.SendAsync(new CreateSampleDataOperation());
+
+        await store.Maintenance.SendAsync(new PutConnectionStringOperation<AiConnectionString>(config.Connection));
+
+        using var session = store.OpenAsyncSession();
+
+        var agent = new AiAgentConfiguration(config.ConnectionStringName,
+            "You are an AI agent of an online shop, helping customers answer queries about that topic only. When talking about orders or products, include the ids as well.");
+
+        agent.Persistence = new AiAgentConfiguration.PersistenceConfiguration
+        {
+            Collection = "Chats",
+            Expires = TimeSpan.FromDays(30)
+        };
+
+        agent.Queries =
+        [
+            AiAgentConfiguration.ToolQuery.Build(
+                    "ProductSearch",
+                    "semantic search the store product catalog",
+                    session.Query<Product>().VectorSearch(v => v.WithText(p => p.Name), v => v.ByText("$query")))
```

Comment:
we should use a lambda here that would accept the session from it.
`session => session.Query(... `

so you won't have to create a new session like this.